legend-engine-xt-daml-model/src/main/java/org/finos/legend/engine/external/format/daml/generation/DAMLGenerationExtension.java

```diff
@@ -0,0 +1,103 @@
+import org.finos.legend.engine.external.shared.format.generations.description.GenerationConfigurationDescription;
+import org.finos.legend.engine.external.shared.format.generations.description.GenerationProperty;
+import org.finos.legend.engine.external.shared.format.imports.description.ImportConfigurationDescription;
+import org.finos.legend.engine.language.pure.compiler.toPureGraph.CompileContext;
+import org.finos.legend.engine.language.pure.compiler.toPureGraph.PureModel;
+import org.finos.legend.engine.language.pure.modelManager.ModelManager;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.PackageableElement;
+import org.finos.legend.pure.generated.Root_meta_pure_generation_metamodel_GenerationConfiguration;
+import org.finos.legend.pure.generated.Root_meta_pure_generation_metamodel_GenerationOutput;
+import org.finos.legend.pure.generated.core_external_language_daml_deprecated_generation;
+
+import java.util.List;
+
+@Deprecated
+public class DAMLGenerationExtension implements GenerationExtension
+{
+    @Override
+    public String getLabel()
+    {
+        return "DAML";
+    }
+
+    @Override
+    public String getKey()
+    {
+        return "daml";
+    }
+
+    @Override
+    public GenerationMode getMode()
+    {
+        return GenerationMode.Schema;
+    }
+
+    @Override
+    public GenerationConfigurationDescription getGenerationDescription()
+    {
+        return new GenerationConfigurationDescription()
+        {
+            @Override
+            public String getLabel()
+            {
+                return DAMLGenerationExtension.this.getLabel();
+            }
+
+            @Override
+            public String getKey()
+            {
+                return DAMLGenerationExtension.this.getKey();
+            }
+
+            @Override
+            public List<GenerationProperty> getProperties(PureModel pureModel)
+            {
+                return FileGenerationDescription.extractGenerationProperties(core_external_language_daml_deprecated_generation.Root_meta_external_language_daml_generation_describeConfiguration__GenerationParameter_MANY_(pureModel.getExecutionSupport()));
+            }
+        };
+    }
+
+    @Override
+    public ImportConfigurationDescription getImportDescription()
+    {
+        return null;
+    }
+
+    @Override
+    public Root_meta_pure_generation_metamodel_GenerationConfiguration defaultConfig(CompileContext context)
+    {
+        return core_external_language_daml_deprecated_generation.Root_meta_external_language_daml_generation_defaultConfig__DAMLConfig_1_(context.pureModel.getExecutionSupport());
+    }
+
+    @Override
+    public Object getService(ModelManager modelManager)
+    {
+        return new DAMLGenerationService(modelManager);
+    }
+
+    @Override
+    public List<Root_meta_pure_generation_metamodel_GenerationOutput> generateFromElement(PackageableElement element, CompileContext compileContext)
+    {
+        return null;
```

Comment:
Want to confirm we don't want to support DAML generation as FileGeneration or SDLC pipeline